dissect/target/loaders/mqtt.py

```diff
@@ -275,25 +271,15 @@ def __init__(self, path: Union[Path, str], **kwargs):
             cls.broker.connect()
             num_peers = int(options.get("peers", 1))
 
-        self.broker = cls.broker
-        self.connection = MQTTConnection(self.broker, self.path)
-        self.peers = self.connection.topo(num_peers)
+        cls.connection = MQTTConnection(cls.broker, path)
+        cls.peers = cls.connection.topo(num_peers)
+        for index, peer in enumerate(cls.peers):
+            yield peer
 
     def map(self, target: Target) -> None:
-        if len(self.peers) == 1 and self.peers[0] == str(self.path):
-            target.path = Path(str(self.path))
-            for disk in self.connection.info():
-                target.disks.add(RawContainer(disk))
-        else:
-            target.mqtt = True
-            vfs = VirtualFilesystem()
-            vfs.map_file_fh(self.PATH, BytesIO("\n".join(self.peers).encode("utf-8")))
-            for index, peer in enumerate(self.peers):
-                vfs.map_file_fh(f"{self.FOLDER}/host{index}-{peer}", BytesIO(peer.encode("utf-8")))
-
-            target.fs.mount("/data", vfs)
-            target.filesystems.add(vfs)
+        for disk in self.connection.info():
+            target.disks.add(RawContainer(disk))
 
     @staticmethod
```

Comment:
Can you put the methods in the order `__init__`, `detect`, `find_all`, `map`? I know it doesn't matter, but it's more consistent with the rest of the loaders.